code/script.js

```diff
@@ -247,47 +251,75 @@ const generateBoard = () => {
 // Randomly select a person from the characters array and set as the value of the variable called secret
 const setSecret = () => {
   secret = charactersInPlay[Math.floor(Math.random() * charactersInPlay.length)]
+//console.log(secret.name)
 }
 
 // This function to start (and restart) the game
 const start = () => {
   // Here we're setting charactersInPlay array to be all the characters to start with
   charactersInPlay = CHARACTERS
+  winOrLose.style.display = "none"
+  board.style.display = "flex"
+  setSecret()
+  generateBoard()
   // What else should happen when we start the game?
 }
 
+start()
+
 // setting the currentQuestion object when you select something in the dropdown
 const selectQuestion = () => {
   const category = questions.options[questions.selectedIndex].parentNode.label
   // This variable stores what option group (category) the question belongs to.
   // We also need a variable that stores the actual value of the question we've selected.
 
-  if (category === 'hair color') {
+const value = questions.value
+
+if (category === 'hair color') {
     currentQuestion = {
       attribute: 'hairColor',
-      // value: ,
-      // 👆 add the value from the input here
-      category: category,
+      value: value,
+      category: category
     }
   } else if (category === 'eye color') {
-    // Set this up your self
+    currentQuestion = {
+      attribute: 'eyeColor',
+      value: value,
+      category: category
+    }
   } else if (category === 'accessories') {
     currentQuestion = {
-      //attribute: ,
-      // 👆 this is the property of the booleans such as smoke, glasses and hat. add the value from the input here
-      value: true, // we're asking if this person wears a hat for exaple, so always true in the question.
-      category: category,
+      attribute: value, //olika värden inom samma kategori, därför sätter vi värde här istället
+       value: true, // we're asking if this person wears a hat for exaple, so always true in the question.
+      category: category
     }
   } else if (category === 'other') {
-    // Set this up your self (should be same structure as above)
+    currentQuestion = {
+      attribute: 'smoker',
+      value: true,
+      category: category
+    }
   }
+  //console.log(currentQuestion.value)
 }
 
+
 // This function should be invoked when you click on 'Find Out'.
+
 const checkQuestion = () => {
   // Compare the currentQuestion with the secret person.
   // See if we should keep or remove people based on that
+  console.log(currentQuestion.attribute)
+ 
+  const charactersToRender = charactersInPlay.filter((characters) => {
+  return characters.value === currentQuestion.value
+
+  })
+
+  generateBoard()
+  
   // Then invoke filterCharacters
+  //filterCharacters()
 }
```

Comment:
Here we want to compare the `currentQuestion` object with our secret person, so the filtering we can do later on in the `filterCharacters()` function. So here we only want to compare the `currentQuestion.value` with the corresponding attribute of the secret person, like so:
`currentQuestion.value === secret[currentQuestion.attribute]`

We can then pass that along when we invoke `filterCharacters()`

Reply:
Check the latest Q/A for guidance on how to approach the `filterCharacter` function's logic :)